src/settings.js

```diff
@@ -0,0 +1,567 @@
+                        },
+                        "onCreate.setTitle": {
+                            "this": "{that}.dom.title",
+                            method: "append",
+                            args: "{settingRow}.model.title"
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.registerNamespace("gpii.app.settings.settingsVisualizer");
+
+    gpii.app.settings.settingsVisualizer.getRowContainerClass = function (markups, containerIndex) {
+        return "." + fluid.stringTemplate(markups.containerClassPrefix, { id: containerIndex });
+    };
+
+    gpii.app.settings.settingsVisualizer.getRowContainerMarkup = function (markups, containerIndex) {
+        // Remove the "." prefix
+        var containerClass = gpii.app.settings.settingsVisualizer.getRowContainerClass(markups, containerIndex).substring(1);
+        return fluid.stringTemplate(markups.container, { containerClass: containerClass });
+    };
+
+    gpii.app.settings.settingsVisualizer.getWidgetMarkup = function (resources, widgetGrade) {
+        //TODO improve; use markup if any
+        return resources[widgetGrade] && resources[widgetGrade].resourceText;
+    };
+
+    /**
+     * Returns the widget grade matching the given GPII setting scheme type.
+     *
+     * @param that {Object} A fluid component containing `widgets` option, e.g. `settingsVisualizer`
+     * @param type {String} GPII setting scheme type
+     * @returns {String} The widget to be used
+     */
+    gpii.app.settings.settingsVisualizer.getWidgetGrade = function (that, type) {
+        var widgetPrefix = that.options.widgets.gradePrefix,
+            widgetGrade,
+            widgetGradesToTypes = that.options.widgets.typesToGrades;
+
+        if (!widgetGradesToTypes) {
+            console.log("`widgetGradesToTypes` missing - Widget grades to types not supplied.");
+        }
+
+        widgetGrade = widgetGradesToTypes[type];
+        if (!widgetGrade) {
+            console.log("Widget " + type + " is not supported.");
+        }
+
+        return widgetPrefix + "." + widgetGrade;
+    };
+
+    // TODO simplify - probably do this sort of mapping somewhere up the chain?
+    gpii.app.settings.settingsVisualizer.getRequiredResources = function (settingsVisualizer, settings) {
+        function appendWidgetResources(resources, settings, settingsVisualizer) {
+            var widgetGrades = settings
+                .map(function (setting) { return setting.type; })
+                .filter(function uniq(setting, idx, settings) { return settings.indexOf(setting) === idx; })
+                .map(
+                    // receives the GPII setting type
+                    gpii.app.settings.settingsVisualizer.getWidgetGrade.bind(null, settingsVisualizer)
+                );
+
+            widgetGrades.forEach(function (widgetGrade) {
+                var template = fluid.defaults(widgetGrade).template;
```

Comment:
In general its better to source such material from an instantiated component rather than defaults - this allows the possibility for interception by later authors. For example, "templates" can be a 3rd argument to this function, - then "getRequiredResources" can be an invoker on the component, and then you will have the space to write a complex definition for the 3rd argument including the templates. 

e.g.

```
getRequiredResources: {
    funcName: "gpii.app.settings.settingsVisualizer.getRequiredResources",
    args: ["{settingsVisualizer}", "{settingsVisualizer}.model.settings", "{settingsVisualizer}.templates"]
}
```
The "exemplars" pattern below shows a route to sourcing this material from components rather than trying to read defaults directly.